src/idl_gen_kotlin.cpp

```diff
@@ -125,6 +127,9 @@ class KotlinGenerator : public BaseGenerator {
       code += "import com.google.flatbuffers.*\n\n";
     }
     code += classcode;
+    if (need_experimental_unsigned_types) {
+      code.insert(code.find("class " + defname), "@ExperimentalUnsignedTypes\n");
```

Comment:
Why do you need to scan through the string until you find your class to add the annotation? This is inefficient.

It would be better for you to add the annotation before generating the class code. Just loop through the fields and check for a single unsigned.